pkg/webhooks/jobset_webhook.go

```diff
@@ -207,6 +207,33 @@ func (j *jobSetWebhook) ValidateCreate(ctx context.Context, obj runtime.Object)
 			allErrs = append(allErrs, fmt.Errorf("invalid replicatedJob name '%s' does not appear in .spec.ReplicatedJobs", rjobName))
 		}
 	}
+
+	// Validate failure policy
+	if js.Spec.FailurePolicy != nil {
+		validOnJobFailureReasons := []string{
```

Comment:
Let's move this to a package level var at the top, no need to reinitialize on every validate call

Reply:
Done.